dcos_test_utils/dcos_api.py

```diff
@@ -264,8 +264,13 @@ def _wait_for_dcos_history_data(self):
             # We have observed cases of the returned JSON being '{}'.
             if 'slaves' in json:
                 # if an agent was removed, it may linger in the history data
-                assert len(json["slaves"]) >= len(self.all_slaves)
-                return True
+                # so simply check that all the agents we expect are present
+                if set(self.all_slaves).intersection(set(json['slaves'])):
```

Comment:
I read this as "as soon as there is one expected agent, return True". Am I misreading?

Reply:
yeah, youre right, I should use subset, not intersection! Yay reviews!